neuralop/layers/recurrent_layers.py

```diff
@@ -0,0 +1,70 @@
+import torch
+import numpy as np
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..layers.spectral_convolution import SpectralConv
+from ..layers.fno_block import FNOBlocks
+
+class RNO_cell(nn.Module):
+    def __init__(self, n_modes, width, output_scaling_factor=None, skip='linear', fft_norm='ortho', factorization=None, separable=False):
+        # output_scaling_factor is provided here as an integer or float
+        super(RNO_cell, self).__init__()
+
+        self.width = width
+
+        scaling_factor = None if not output_scaling_factor else [output_scaling_factor]
+
+        # Some output_scaling_factors are None to super-resolution purposes. We use the hidden representation in the scaled size always (it's initialized that way),
+        # so we only need to scale the dimensions of f1, f3, and f5, which act on x (original dimensionality).
+        self.f1 = FNOBlocks(width, width, n_modes, output_scaling_factor=scaling_factor, fno_skip='linear', fft_norm=fft_norm, factorization=factorization, separable=separable)
+        self.f2 = FNOBlocks(width, width, n_modes, output_scaling_factor=None, fno_skip='linear', fft_norm=fft_norm, factorization=factorization, separable=separable)
+        self.f3 = FNOBlocks(width, width, n_modes, output_scaling_factor=scaling_factor, fno_skip='linear', fft_norm=fft_norm, factorization=factorization, separable=separable)
+        self.f4 = FNOBlocks(width, width, n_modes, output_scaling_factor=None, fno_skip='linear', fft_norm=fft_norm, factorization=factorization, separable=separable)
+        self.f5 = FNOBlocks(width, width, n_modes, output_scaling_factor=scaling_factor, fno_skip='linear', fft_norm=fft_norm, factorization=factorization, separable=separable)
+        self.f6 = FNOBlocks(width, width, n_modes, output_scaling_factor=None, fno_skip='linear', fft_norm=fft_norm, factorization=factorization, separable=separable)
+
+        self.b1 = nn.Parameter(torch.normal(torch.tensor(0.),torch.tensor(1.))) # constant bias terms
+        self.b2 = nn.Parameter(torch.normal(torch.tensor(0.),torch.tensor(1.)))
+        self.b3 = nn.Parameter(torch.normal(torch.tensor(0.),torch.tensor(1.)))
+    
+    def forward(self, x, h):
+        z = torch.sigmoid(self.f1(x) + self.f2(h) + self.b1)
+        r = torch.sigmoid(self.f3(x) + self.f4(h) + self.b2)
+        h_hat = F.selu(self.f5(x) + self.f6(r * h) + self.b3) # selu for regression problem
+
+        h_next = (1. - z) * h + z * h_hat
+
+        return h_next
+
+class RNO_layer(nn.Module):
```

Comment:
Can we add docstrings here, also a reference for the paper so users can cite and short description of what it does